educe/graph.py

```diff
@@ -502,16 +502,15 @@ def cdu_members(self, cdu, deep=False):
         that are members of (members of ..) this CDU.
         """
 
+        hyperedge = self.edgeform(cdu)
+        members = set(self.links(hyperedge))
+
         if deep:
-            members = set()
-            for m in self.cdu_members(cdu):
-                members.add(m)
+            for m in list(members):
```

Comment:
You can already iterate over the set form of `members`, so why convert into a list?
If this has a noticeable impact on speed for our typical use cases, we can keep this but you should add a comment stating that this conversion to list is just an optimization (as opposed to a semantically motivated conversion).